### PR TITLE
Fix CPU usage and exit normally

### PR DESCRIPTION
**Pull request by xou130**

### CODE REVIEW

**Comment by samoylenko** [comment]
Thank you @xou130, the change looks good to me, but I am not qualified to check and/or test. Requested review by @m417z , hope its ok. If he's busy, I'll merge in a day or two

**Comment by samoylenko** [comment]
Hey @xou130 , I noticed a pair of commits that may be not intended. Would recommend rebasing to the updated master.

image

File: service.c [hunk 1]
```diff
@@ -73,8 +73,9 @@ VOID ServiceStart(DWORD dwArgc, LPTSTR *lpszArgv)
 		goto cleanup;
 	}
 
-	while(1)
+	while(ssStatus.dwCurrentState == SERVICE_RUNNING)
```
Comment (m417z) [inline service.c:76]: Does it actually work? I didn't check, but I think that it's better to wait for `hServerStopEvent`. That's also what is done in the [Microsoft example](https://github.com/microsoft/Windows-classic-samples/blob/1d363ff4bd17d8e20415b92e2ee989d615cc0d91/Samples/Win7Samples/winbase/service/Simple.c).

File: service.c [hunk 2]
```diff
@@ -73,8 +73,9 @@ VOID ServiceStart(DWORD dwArgc, LPTSTR *lpszArgv)
 		goto cleanup;
 	}
 
-	while(1)
+	while(ssStatus.dwCurrentState == SERVICE_RUNNING)
 	{
+	    Sleep(1000);
```
Comment (m417z) [inline service.c:78]: You're mixing tabs with spaces, here and in other places.

File: service.c [hunk 3]
```diff
@@ -83,11 +84,12 @@ VOID ServiceStart(DWORD dwArgc, LPTSTR *lpszArgv)
 	{
 		CloseHandle(hServerStopEvent);
 	}
-
 	if(hEvents[1])
 	{
 		CloseHandle(hEvents[1]);
 	}
+    ReportStatusToSCMgr(SERVICE_STOPPED, NO_ERROR, 0);
+    exit(0);
```
Comment (m417z) [inline service.c:92]: `ReportStatusToSCMgr(SERVICE_STOPPED, ...);` is called in `service_main`, why call it here? And why is `exit(0);` necessary?

File: service.c [hunk 4]
```diff
@@ -143,14 +145,13 @@ int main(int argc, char **argv)
 	{
 		AddToMessageLog(TEXT("StartServiceCtrlDispatcher failed."));
 	}
-
+    system("pause");
```
Comment (m417z) [inline service.c:148]: Usually, the debugger is responsible for leaving the terminal window open. I wouldn't add this.

**Comment by m417z** [comment]
Frankly, I just found this repo handy for quickly testing some code running as a service. I have no strong feelings about the changes, but they don't look so correct to me, as I pointed out in the comments.

**Comment by samoylenko** [comment]
> Frankly, I just found this repo handy for quickly testing some code running as a service. I have no strong feelings about the changes, but they don't look so correct to me, as I pointed out in the comments.

Thank you so much @m417z , very much appreciate the prompt and excellent review.

**Comment by samoylenko** [comment]
Thank you for your contribution @xou130 , your help is appreciated. But I agree with @m417z , including what he said about the this repository. I decided to freeze this repository (and others that I keep only as an archive). Will update the README accordingly, and will check if GitHub allows to somehow block/note that on any new pull requests. 

I think that the best way forward with this code updates is if you'd like to own the fork. I will be happy to add a references to it here.

**Pull request closed without merging**